azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -660,8 +660,9 @@ def _check_processes_in_agent_cgroup(self):
                 systemd_run_commands.update(self._cgroups_api.get_systemd_run_commands())
 
                 for process in agent_cgroup:
+                    agent_cgroup_proc_names.append(self.__format_process(process))
                     # Note that the agent uses systemd-run to start extensions; systemd-run belongs to the agent cgroup, though the extensions don't.
-                    if process in (daemon, extension_handler) or process in systemd_run_commands:
+                    if process in (daemon, extension_handler) or process in systemd_run_commands or self._check_systemd_run_process(process):
```

Comment:
why do we need this check here? the check on 668/669 should capture the systemd-run commands

Reply:
The check on 668/669 only works if parent process found for systemd-run

I have seen a case where the shell process gone and new process is started to execute the system-run arguments. We have discussed this last month when we found it in dcr.

```

2022-11-06T05:54:14.613791Z INFO MonitorHandler ExtHandler [CGW] Disabling resource usage monitoring. Reason: Check on cgroups failed:
[CGroupsException] The agent's cgroup includes unexpected processes: ['[PID: 2295] /bin/sh\x00-c\x00systemd-run --property=CPUAccounting=no --property=Me'] 
```